crates/sui-node/src/lib.rs

```diff
@@ -388,11 +417,40 @@ impl SuiNode {
             .ok_or_else(|| anyhow::anyhow!("Transaction Orchestrator is not enabled in this node."))
     }
 
-    //TODO watch/wait on all the components
-    pub async fn wait(self) -> Result<()> {
-        self.grpc_server.await??;
-
-        Ok(())
+    pub async fn monitor_reconfiguration(mut self) -> Result<()> {
+        loop {
+            let next_epoch = self
+                .reconfig_channel
+                .1
+                .recv()
+                .await
+                .expect("Reconfiguration channel was closed unexpectedly.");
+            info!(
+                ?next_epoch,
+                "Received reconfiguration signal. About to reconfigure the system."
+            );
+            let new_committee = self
```

Comment:
Do we need compare SuiSystemState::epoch and next_epoch and assert they are equal?